extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -250,40 +273,77 @@ void generateConfigClasses(
     }
 
     @BuildStep
-    void beanConfigClasses(
+    void registerConfigMappingsBean(
+            BeanRegistrationPhaseBuildItem beanRegistration,
             List<ConfigClassBuildItem> configClasses,
-            BeanRegistrationPhaseBuildItem beanRegistrationPhase,
             CombinedIndexBuildItem combinedIndex,
-            BuildProducer<BeanConfiguratorBuildItem> beanConfigurationRegistry) {
+            BuildProducer<BeanConfiguratorBuildItem> beanConfigurator) {
+
+        Set<ConfigClassBuildItem> configMappings = new HashSet<>();
+        for (InjectionPointInfo injectionPoint : beanRegistration.getInjectionPoints()) {
+            Type type = injectionPoint.getType();
+            for (ConfigClassBuildItem configClass : configClasses) {
```

Comment:
This does not seem to be very efficient for an application with a large number of beans and injection points. I'd probably collect all config classes first, store them in a `Set<DotName> configClasses` and then simply test `configClasses.contains(type.name())` for each injection point.

Reply:
Hm, I'm talking about replacement of `type.name().equals(DotName.createSimple(configClass.getConfigClass().getName())`... so no prefix is involved here.

Reply:
Yes. I'm already changing the code. I'll send an update soon.

Reply:
I'm sorry but I don't see any improvement here. You're still iterating over all `configClasses` for each injection point.
 
Also what's the reason to store `ConfigClassBuildItem` in a set? I mean the class itself does not define `equals()`/`hashCode()` so a list would be probably more appropriate...

BTW we could also skip processing completely if `configClasses.isEmpty()`.

Reply:
Ah sorry, nvm. I've changed the validate* methods but not this one.